### PR TITLE
Split services into smaller chunks

### PR DESCRIPTION
**Pull request by rapidddenis**
solves https://github.com/etherisc/gif-next/issues/193

### CODE REVIEW

File: contracts/instance/base/ComponentServiceBase.sol [hunk 1]
```diff
@@ -43,72 +28,28 @@ abstract contract ComponentServiceBase is Service {
...
+    function getRegistryService() public view virtual returns (IRegistryService) {
+        address service = getRegistry().getServiceAddress("RegistryService", getMajorVersion());
+        return IRegistryService(service);
     }
 
     function getInstanceService() public view returns (InstanceService) {
         return InstanceService(getRegistry().getServiceAddress("InstanceService", getMajorVersion()));
     }
 
-    function register(address componentAddress)
```
Comment (doerfli) [inline contracts/instance/base/ComponentServiceBase.sol:64]: Why did you move the `register` function back into each service? we wanted to have it centralized to ensure the common functions - especially the component owner role checks and the assignment which role is required for which service -  are always called.

File: contracts/instance/service/PolicyService.sol [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+
+import {IService} from "../../shared/IService.sol";
+import {Service} from "../../shared/Service.sol";
+import {ComponentServiceBase} from "../base/ComponentServiceBase.sol";
+import {IPolicyService} from "./IPolicyService.sol";
+import {InstanceReader} from "../InstanceReader.sol";
+import {IPoolService} from "./PoolService.sol";
+import {POOL_SERVICE_NAME} from "./PoolService.sol";
+
+string constant PRODUCT_SERVICE_NAME = "ProductService";
```
Comment (doerfli) [inline contracts/instance/service/PolicyService.sol]: rename or remove this line to avoid duplicates.

File: contracts/instance/base/ComponentServiceBase.sol (hunk 1 above)
Comment (rapidddenis) [reply contracts/instance/base/ComponentServiceBase.sol:64]: register() is special to product, pool and distribution service, while policy and bundle service have createApplication() and creatBundle()

File: contracts/instance/service/PolicyService.sol (hunk 2 above)
Comment (rapidddenis) [reply contracts/instance/service/PolicyService.sol]: fixed

File: scripts/libs/services.ts [hunk 3]
```diff
@@ -25,6 +25,10 @@ export type ServiceAddresses = {
     productServiceNftId: string,
     productService: ProductService,
     productServiceManagerAddress: AddressLike,
+    policyServiceAddress: AddressLike,
```
[the remark this answers is not in the document]
Comment (rapidddenis) [reply scripts/libs/services.ts:28]: fixed

File: test_forge/TestProduct.t.sol [hunk 4]
```diff
@@ -26,7 +26,7 @@ import {BundleManager} from "../contracts/instance/BundleManager.sol";
 contract TestProduct is TestGifBase {
     using NftIdLib for NftId;
 
-    function test_Product_SetFees() public {
+    function test_SetFees() public {
```
[the remark this answers is not in the document]
Comment (rapidddenis) [reply test_forge/TestProduct.t.sol]: fixed

File: scripts/libs/services.ts [hunk 5]
```diff
@@ -137,14 +160,18 @@ export async function deployAndRegisterServices(owner: Signer, registry: Registr
...
+        policyServiceManagerAddress,
     };
 }
 
 const DISTRIBUTION_REGISTRAR_ROLE = 1000;
-// const POLICY_REGISTRAR_ROLE = 1100;
 const BUNDLE_REGISTRAR_ROLE = 1200;
 const POOL_REGISTRAR_ROLE = 1300;
 const PRODUCT_REGISTRAR_ROLE = 1400;
+const POLICY_REGISTRAR_ROLE = 1100;
```
[the remark this answers is not in the document]
Comment (rapidddenis) [reply scripts/libs/services.ts:203]: fixed

File: scripts/deploy_all.ts [hunk 6]
```diff
@@ -156,6 +156,12 @@ function printAddresses(
     addresses += `productServiceManagerAddress: ${services.productServiceManagerAddress}\n`;
     addresses += `productServiceAddress: ${services.productServiceAddress}\n`;
     addresses += `productServiceNftId: ${services.productServiceNftId}\n`;
+    addresses += `policyServiceManagerAddress: ${services.policyServiceManagerAddress}\n`;
+    addresses += `policytServiceAddress: ${services.policyServiceAddress}\n`;
```
Comment (doerfli) [inline scripts/deploy_all.ts:160]:
```suggestion
    addresses += `policyServiceAddress: ${services.policyServiceAddress}\n`;
```

**Review by doerfli — APPROVED** [review]
